elastic/kibana
# [Osquery] Add telemetry for packs and saved queries

## Summary

Collecting telemetry of live queries.
Took it from https://github.com/elastic/kibana/pull/115180

See `sender.ts`
To test:
Run live query

For event sender:
Add to `kibana.dev.yml` to opt in telemetry and see debug logs:
```
telemetry.optIn: true
logging.loggers:
 - name: plugins.osquery
   level: debug
```
In case of success/failure of upgrades, the debug log should include events sent to telemetry v3 endpoint.

```
[2022-01-09T22:08:02.359+01:00][DEBUG][plugins.osquery.telemetry_events] Telemetry URL: https://telemetry-staging.elastic.co/v3/send/osquery-live-queries
[2022-01-09T22:08:02.360+01:00][DEBUG][plugins.osquery.telemetry_events] [{"event_source":"osquery_app_live_query","table":[{"name":"osquery_info","columns":["*"]},{"name":"processes","columns":["resident_size","user_time","system_time"]},{"name":"time","columns":["minutes"]}],"agent_selection":{"agents":1,"all_agents_selected":false,"platforms_selected":["darwin"],"policies":1}},{"event_source":"osquery_app_live_query","table":[{"name":"osquery_info","columns":["*"]},{"name":"processes","columns":["resident_size","user_time","system_time"]},{"name":"time","columns":["minutes"]}],"agent_selection":{"agents":1,"all_agents_selected":false,"platforms_selected":["darwin"],"policies":1}}]
[2022-01-09T22:08:02.717+01:00][DEBUG][plugins.osquery.telemetry_events] Events sent!. Response: 200 {"status":"ok"}
```
